tests/expose_test.go

```diff
@@ -331,89 +557,177 @@ var _ = Describe("[rfe_id:253][crit:medium][vendor:cnv-qe@redhat.com][level:comp
+		}
+
+		createStoppedVM := func(virtClient kubecli.KubevirtClient, namespace string) error {
+			By("Creating an VM object")
+			template := newLabeledVMI("vm", virtClient, false)
+			vm = tests.NewRandomVirtualMachine(template, false)
+
+			By("Creating the VM")
+			vm, err = virtClient.VirtualMachine(namespace).Create(vm)
+			return err
+		}
+
+		startVMWithServer := func(protocol string, port int) *v1.VirtualMachineInstance {
+			By("Calling the start command on the stopped VM")
+			vmi := startVMIFromVMTemplate(vm.GetName(), vm.GetNamespace())
+			if vmi == nil {
+				return nil
+			}
+			tests.GenerateHelloWorldServer(vmi, port, protocol)
+			return vmi
+		}
+
+		cleanupVM := func(virtClient kubecli.KubevirtClient, namespace string, vmName string) error {
+			By("Deleting the VM ...")
+			err := virtClient.VirtualMachine(vm.GetNamespace()).Delete(vm.GetName(), &k8smetav1.DeleteOptions{})
+			if err != nil {
+				return err
+			}
+
+			err = wait.PollImmediate(time.Second, 2*time.Minute, func() (done bool, err error) {
+				_, err = virtClient.VirtualMachineInstance(vm.GetNamespace()).Get(vm.GetName(), &k8smetav1.GetOptions{})
+				return errors.IsNotFound(err), nil
+			})
+			if err != nil {
+				return fmt.Errorf("error deleting VMI: %v", err)
+			}
+
+			err = wait.PollImmediate(time.Second, 2*time.Minute, func() (done bool, err error) {
+				_, err = virtClient.VirtualMachine(vm.GetNamespace()).Get(vm.GetName(), &k8smetav1.GetOptions{})
+				return errors.IsNotFound(err), nil
+			})
+			if err != nil {
+				return fmt.Errorf("error deleting VM template: %v", err)
+			}
+			return err
+		}
+
+		doesVMExist := func(vmName string, namespace string) bool {
+			_, err = virtClient.VirtualMachine(namespace).Get(vmName, &k8smetav1.GetOptions{})
+			return !errors.IsNotFound(err)
+		}
 
 		Context("Expose a VM as a ClusterIP service.", func() {
-			It("[test_id:1538][label:masquerade_binding_connectivity]Connect to ClusterIP service that was set when VM was offline.", func() {
+			tests.BeforeAll(func() {
+				tests.BeforeTestCleanup()
+			})
+
+			BeforeEach(func() {
+				Expect(createStoppedVM(virtClient, tests.NamespaceTestDefault)).To(Succeed(), "should create a stopped VM.")
+			})
+
+			BeforeEach(func() {
+				vmExposeArgs = []string{
+					expose.COMMAND_EXPOSE,
+					"virtualmachine", "--namespace", vm.GetNamespace(), vm.GetName(),
+					"--port", servicePort, "--name", serviceName, "--target-port", strconv.Itoa(testPort),
+				}
+			})
+
+			AfterEach(func() {
+				Expect(jobs).NotTo(BeEmpty(), "a successful test must have stored a way to delete the batchv1.Job entity")
+				Expect(cleanupJobs(jobs)).To(BeEmpty(), "should be able to delete the multiple k8sv1.`Job` entities")
+			})
+
+			AfterEach(func() {
+				Expect(cleanupService(serviceName, tests.NamespaceTestDefault)).To(Succeed(), "should be able to delete the k8sv1.Service entity")
+			})
+
+			AfterEach(func() {
+				if doesVMExist(vm.GetName(), vm.GetNamespace()) {
```

Comment:
this vm.Name deref is buggy,
if the vm doesnt exist its a nil ptr deref
for example if 
https://github.com/kubevirt/kubevirt/pull/4272/files#diff-bf4f2c831a0b99974eedbd27880b2ebc7d3e68e63a5a06613be323930225d1d6R642 is failed

Reply:
If that fails, we don't have a VM, and the test is already failed.

Meaning there's nothing to cleanup, and nothing to test. From the perspective of the test framework, there's nothing to do.

Reply:
but there is a panic, silent one but it exists, agree?
i am not sure we are fine with silent panics even that they are silent

Reply:
I agree that in the case you present it will error - but we will **never** see the error. The test will already be failed, and there is nothing to cleanup.

I really don't understand why you want to add more code to control a scenario that you will never face (the error is eaten away, and the real error presented - i.e. why did the VM fail to be created in the first place).

Reply:
just because we might want the code to not swallow errors as this one, especially when the solution is easy imo, 
setting the vm to nil and checking if its nil before deref, this is the right thing to do imo

Reply:
It is not us that are swallowing the error, it is the test framework - which is specified to work like that. 

You keep pointing out that "the solution is easy". And I keep pointing out that there is no problem to fix. Unless you come up with a real reason why there is a problem, the fact that the solution is easy is irrelevant (since there is no problem).

Reply:
masking the error is a problem imo, one might consider it minor, one can consider it not problem at all,
i do consider it a minor problem, and when i see there is an easy and right solution, cleaning a variable that might be used wrongly, the problem gets little bigger in my perspective

up to you what to do, this is my opinion

Reply:
> masking the error is a problem imo, one might consider it minor, one can consider it not problem at all,

Once again: I am not masking any error. The test framework is. You're free to open a bug to ginkgo explaining you think they're masking errors, and asking their behavior to change. 

> i do consider it a minor problem, and when i see there is an easy and right solution, cleaning a variable that might be used wrongly, the problem gets little bigger in my perspective
> 
> up to you what to do, this is my opinion

You're entitled to it.

Reply:
> > masking the error is a problem imo, one might consider it minor, one can consider it not problem at all,
> 
> Once again: I am not masking any error. The test framework is. You're free to open a bug to ginkgo explaining you think they're masking errors, and asking their behavior to change.

We still can / might want to code such it wont happen, generally speaking.

> 
> > i do consider it a minor problem, and when i see there is an easy and right solution, cleaning a variable that might be used wrongly, the problem gets little bigger in my perspective
> > up to you what to do, this is my opinion
> 
> You're entitled to it.